BeagleBone_black/src/navigation/main.cpp

```diff
@@ -41,6 +44,14 @@ void Main::run()
 
   *last_readings = data_.getSensorsData();  // TODO(Brano): Make sure data_ is properly initd
   while (1) {
+    // State updates
+    State current_state = data_.getStateMachineData().current_state;
+    if (current_state != State::kIdle
+        && current_state != State::kReady
+        && nav_.getState() != NavigationState::kOperational
+        && nav_.getState() != NavigationState::kCriticalFailure)
+      nav_.finishCalibration();
```

Comment:
the return value is not used, how do you know if you called the barrier?

Can you separate the calibration loop (and this exit check) from the main operational loop. Those can be two separate whiles, so that it is clear the code cannot start calibrating again after already being operational.

Reply:
It turns out I don't really care. If I didn't call the barrier, then I'm still calibrating. I thought it might be useful when I wrote the function, but it turned out, `finishCalibration()` could have been declared to return `void` as well.